### PR TITLE
Update Business::ISBN version requirements

### LINKED ISSUE #83: t/urn-isbn.t may fail
Seen on some of my smokers:
```
Use of uninitialized value in string eq at t/urn-isbn.t line 19.
Use of uninitialized value in numeric eq (==) at t/urn-isbn.t line 22.
Use of uninitialized value in numeric eq (==) at t/urn-isbn.t line 25.
Use of uninitialized value in string eq at t/urn-isbn.t line 28.
Use of uninitialized value in string eq at t/urn-isbn.t line 34.
Use of uninitialized value in string eq at t/urn-isbn.t line 40.
t/urn-isbn.t .............. 
Failed 7/13 subtests 
```
It's possible that this is caused by some version of Business::ISBN; especially if the latest version is installed, then this does not seem to happen.

### PR DESCRIPTION
**Pull request by oalders**
Closes #83 and borrows patch from #84.

### CODE REVIEW

File: lib/URI/urn/isbn.pm [hunk 1]
```diff
@@ -11,13 +11,13 @@ use Carp qw(carp);
 
 BEGIN {
     require Business::ISBN;
-    
+
     local $^W = 0; # don't warn about dev versions, perl5.004 style
-    warn "Using Business::ISBN version " . Business::ISBN->VERSION . 
+    warn "Using Business::ISBN version " . Business::ISBN->VERSION .
         " which is deprecated.\nUpgrade to Business::ISBN version 2\n"
```
Comment (skaji) [inline lib/URI/urn/isbn.pm:17]: version 2 -> version 3.005

Comment (oalders) [reply lib/URI/urn/isbn.pm]: Good catch. Thanks!

**Review by skaji — APPROVED** [review]
LGTM

**Comment by oalders** [comment]
Thanks, @skaji!

**Pull request merged**